feathr_project/test/test_azure_spark_e2e.py

```diff
@@ -443,17 +465,54 @@ def test_feathr_materialize_with_time_partition_pattern():
+    client_consumer.materialize_features(settings_tpp)
     # assuming the job can successfully run; otherwise it will throw exception
     client_consumer.wait_job_to_finish(timeout_sec=Constants.SPARK_JOB_TIMEOUT_SECONDS)
 
     # download result and just assert the returned result is not empty
     # by default, it will write to a folder appended with date
-    res_df = get_result_df(client_consumer, "avro", output_path_tpp + "/df0/daily/2020/05/20")
+    res_df = get_result_df(client_consumer, "avro", output_path_tpp + "/df0/daily/2020/05/21")
+    assert res_df.shape[0] > 0
+   
+@pytest.mark.skip(reason="Will add back when updated scala jar is applied")    
+def test_feathr_materialize_with_time_partition_pattern_postfix_path():  
+    test_workspace_dir = Path(
+        __file__).parent.resolve() / "test_user_workspace"
+   
+    client_producer: FeathrClient = basic_test_setup(os.path.join(test_workspace_dir, "feathr_config.yaml"))
+    if client_producer.spark_runtime == 'databricks':
+        output_path = 'dbfs:/timePartitionPattern_test'
+    else:
+        output_path = 'abfss://feathrazuretest3fs@feathrazuretest3storage.dfs.core.windows.net/timePartitionPattern_test'
+    # test 'timePartitionPattern' with postfix path
+    client_consumer_pf: FeathrClient = time_partition_pattern_test_setup(os.path.join(test_workspace_dir, "feathr_config.yaml"), output_path+'/df0/daily', postfix_path='postfixPath')
+
+    backfill_time_pf = BackfillTime(start=datetime(
+        2020, 5, 11), end=datetime(2020, 5, 11), step=timedelta(days=1))
+
+    now = datetime.now()
+    if client_consumer_pf.spark_runtime == 'databricks':
+        output_path_pf = ''.join(['dbfs:/feathrazure_cijob_materialize_offline_','_', str(now.minute), '_', str(now.second), ""])
+    else:
+        output_path_pf = ''.join(['abfss://feathrazuretest3fs@feathrazuretest3storage.dfs.core.windows.net/demo_data/feathrazure_cijob_materialize_offline_','_', str(now.minute), '_', str(now.second), ""])
+    offline_sink_pf = HdfsSink(output_path=output_path_pf)
+    settings_pf = MaterializationSettings("nycTaxiTable",
+                                       sinks=[offline_sink_pf],
+                                       feature_names=[
+                                           "f_loc_avg_output", "f_loc_max_output"],
+                                       backfill_time=backfill_time_pf)
+    client_consumer_pf.materialize_features(settings_pf)
+    # assuming the job can successfully run; otherwise it will throw exception
+    client_consumer_pf.wait_job_to_finish(timeout_sec=Constants.SPARK_JOB_TIMEOUT_SECONDS)
+
+    # download result and just assert the returned result is not empty
+    # by default, it will write to a folder appended with date
+    res_df = get_result_df(client_consumer_pf, "avro", output_path_pf + "/df0/daily/2020/05/11")
     assert res_df.shape[0] > 0
     
-    
+   
```

Comment:
Unrelated with this PR - maybe we should remove those:

if __name__ == "__main__":
    test_feathr_materialize_to_aerospike()
    test_feathr_get_offline_features_to_sql()
    test_feathr_materialize_to_cosmosdb()

Reply:
Moved these test cases to another file and didn't have this problem any more